.github/workflows/build.yaml

```diff
@@ -0,0 +1,72 @@
+name: build
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.6, 3.7, 3.8, 3.9]
+      fail-fast: true
+
+    name: Lint on Python ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Lint
+      run: |
+        pip install flake8 flake8-isort flake8-quotes
+        flake8 .
+  test:
+    needs: [lint]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.6, 3.7, 3.8, 3.9]
+      fail-fast: true
+
+    name: Test on Python ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Test
+      run: |
+        pip install '.'
+        pip install mock pytest pytest-cov codecov coverage
+        pytest -xv --cov ntplib --cov-report term-missing test_ntplib.py
+
+    - name: Upload coverage report to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+  publish:
+    needs: [test]
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 2.7
+    - name: Build package
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+      run: |
+        pip install setuptools wheel
+        python setup.py sdist bdist_wheel
+    - name: Publish package
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
note: token needs to be generated on PYPI for this package and be added on github secrets for this repo